<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>deconvR : Simulation and Deconvolution of Omic Profiles • deconvR</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="deconvR : Simulation and Deconvolution of Omic Profiles">
<meta property="og:description" content="deconvR">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">deconvR</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">1.4.4</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/deconvRVignette.html">deconvR : Simulation and Deconvolution of Omic Profiles</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/BIMSBbioinfo/deconvR/" class="external-link">
    <span class="fab fa-github fa-lg"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>deconvR : Simulation and Deconvolution of Omic
Profiles</h1>
            
            <h4 data-toc-skip class="date">2022-12-05</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/BIMSBbioinfo/deconvR/blob/HEAD/vignettes/deconvRVignette.Rmd" class="external-link"><code>vignettes/deconvRVignette.Rmd</code></a></small>
      <div class="hidden name"><code>deconvRVignette.Rmd</code></div>

    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Recent studies associated the differences of cell-type proportions
may be correlated to certain phenotypes, such as cancer. Therefore, the
demand for the development of computational methods to predict cell type
proportions increased. Hereby, we developed <code>deconvR</code>, a
collection of functions designed for analyzing deconvolution of the bulk
sample(s) using an atlas of reference omic signature profiles and a
user-selected model. We wanted to give users an option to extend their
reference atlas. Users can create new reference atlases using
<code>findSignatures</code> or extend their atlas by adding more cell
types. Additionally, we included <code>BSMeth2Probe</code> to make
mapping whole-genome bisulfite sequencing data to their probe IDs
easier. So users can map WGBS methylation data (as in
<strong>methylKit</strong> or <strong>GRanges</strong> object format) to
probe IDs, and the results of this mapping can be used as the bulk
samples in the deconvolution. We also included a comprehensive DNA
methylation atlas of 25 different cell types to use in the main function
<code>deconvolute</code>. <code>deconvolute</code> allows the user to
specify their desired deconvolution model (non-negative least squares
regression, support vector regression, quadratic programming, or robust
linear regression), and returns a dataframe which contains predicted
cell-type proportions of bulk methylation profiles, as well as partial
R-squared values for each sample.</p>
<p>As an another option, users can generate a simulated table of a
desired number of samples, with either user-specified or random origin
proportions using <code>simulateCellMix</code>.
<code>simulateCellMix</code> returns a second data frame called
<code>proportions</code>, which contains the actual cell-type
proportions of the simulated sample. It can be used for testing the
accuracy of the deconvolution by comparing these actual proportions to
the proportions predicted by <code>deconvolute</code>.</p>
<p><code>deconvolute</code> returns partial R-squares, to check if
deconvolution brings advantages on top of the basic bimodal profiles.
The reference matrix usually follows a bimodal distribution in the case
of methylation, and taking the average of the rows of methylation matrix
might give a pretty similar profile to the bulk methylation profile you
are trying to deconvolute. If the deconvolution is advantageous, partial
R-squared is expect to be high.</p>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>The deconvR package can be installed from Bioconductor with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/ns-load.html" class="external-link">requireNamespace</a></span><span class="op">(</span><span class="st">"BiocManager"</span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/BiocManager/man/install.html" class="external-link">install</a></span><span class="op">(</span><span class="st">"deconvR"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<div class="section level3">
<h3 id="comprehensive-human-methylome-reference-atlas">Comprehensive Human Methylome Reference Atlas<a class="anchor" aria-label="anchor" href="#comprehensive-human-methylome-reference-atlas"></a>
</h3>
<p>The comprehensive human methylome reference atlas created by Moss et
al. <a href="#fn1" class="footnote-ref" id="fnref1"><sup>1</sup></a> can
be used as the reference atlas parameter for several functions in this
package. This atlas was modified to remove duplicate CpG loci before
being included in the package as the dataframe. The dataframe is
composed of 25 human cell types and roughly 6000 CpG loci, identified by
their Illumina Probe ID. For each cell type and CpG locus, a methylation
value between 0 and 1 is provided. This value represents the fraction of
methylated bases of the CpG locus. The atlas therefore provides a unique
methylation pattern for each cell type and can be directly used as
<code>reference</code> in <code>deconvolute</code> and
<code>simulateCellMix</code>, and <code>atlas</code> in
<code>findSignatures</code>. Below is an example dataframe to illustrate
the <code>atlas</code> format.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/BIMSBbioinfo/deconvR" class="external-link">deconvR</a></span><span class="op">)</span> </span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HumanCellTypeMethAtlas"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">HumanCellTypeMethAtlas</span><span class="op">[</span>,<span class="fl">1</span><span class="op">:</span><span class="fl">5</span><span class="op">]</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          IDs Monocytes_EPIC B.cells_EPIC CD4T.cells_EPIC NK.cells_EPIC</span></span>
<span><span class="co">## 1 cg08169020         0.8866       0.2615          0.0149        0.0777</span></span>
<span><span class="co">## 2 cg25913761         0.8363       0.2210          0.2816        0.4705</span></span>
<span><span class="co">## 3 cg26955540         0.7658       0.0222          0.1492        0.4005</span></span>
<span><span class="co">## 4 cg25170017         0.8861       0.5116          0.1021        0.4363</span></span>
<span><span class="co">## 5 cg12827637         0.5212       0.3614          0.0227        0.2120</span></span>
<span><span class="co">## 6 cg19442545         0.2013       0.1137          0.0608        0.0410</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="illumina-infinium-methylationepic-v1-0-b5-manifest-probes-hg38">Illumina Infinium MethylationEPIC v1.0 B5 Manifest Probes
(hg38)<a class="anchor" aria-label="anchor" href="#illumina-infinium-methylationepic-v1-0-b5-manifest-probes-hg38"></a>
</h3>
<p>The <strong>GRanges</strong> object
<code>IlluminaMethEpicB5ProbeIDs</code> contains the Illumina probe IDs
of 400000 genomic loci (identified using the “seqnames”, “ranges”, and
“strand” values). This object is based off of the Infinium
MethylationEPIC v1.0 B5 Manifest data. Unnecessary columns were removed
and rows were truncated to reduce file size before converting the file
to a <strong>GRanges</strong> object. It can be used directly as
<code>probe_id_locations</code> in <code>BSmeth2Probe</code>.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"IlluminaMethEpicB5ProbeIDs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">IlluminaMethEpicB5ProbeIDs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 6 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames              ranges strand |          ID</span></span>
<span><span class="co">##          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##   [1]    chr19     5236005-5236006      + |  cg07881041</span></span>
<span><span class="co">##   [2]    chr20   63216298-63216299      + |  cg18478105</span></span>
<span><span class="co">##   [3]     chr1     6781065-6781066      + |  cg23229610</span></span>
<span><span class="co">##   [4]     chr2 197438742-197438743      - |  cg03513874</span></span>
<span><span class="co">##   [5]     chrX   24054523-24054524      + |  cg09835024</span></span>
<span><span class="co">##   [6]    chr14   93114794-93114795      - |  cg05451842</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="example-workflow-for-whole-genome-bisulfate-sequencing-data">Example Workflow For Whole Genome Bisulfate Sequencing Data<a class="anchor" aria-label="anchor" href="#example-workflow-for-whole-genome-bisulfate-sequencing-data"></a>
</h2>
<div class="section level3">
<h3 id="expanding-reference-atlas">Expanding Reference Atlas<a class="anchor" aria-label="anchor" href="#expanding-reference-atlas"></a>
</h3>
<p>As mentioned in the introduction section, users can extend their
reference atlas to incorporate new data. Or may combine different
reference atlases to construct a more comprehensive one. This can be
done using the <code>findSignatures</code> function. In this example,
since we don’t have any additional reference atlas, we will add
simulated data as a new cell type to reference atlas for example
purposes. First, ensure that <code>atlas</code> (the signature matrix to
be extended) and <code>samples</code> (the new data to be added to the
signature matrix) are compliant with the function requirements. Below
illustrates the <code>samples</code> format.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulateCellMix.html">simulateCellMix</a></span><span class="op">(</span><span class="fl">3</span>,reference <span class="op">=</span> <span class="va">HumanCellTypeMethAtlas</span><span class="op">)</span><span class="op">$</span><span class="va">simulated</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          IDs   Sample 1   Sample 2   Sample 3</span></span>
<span><span class="co">## 1 cg08169020 0.17529579 0.18305686 0.16494031</span></span>
<span><span class="co">## 2 cg25913761 0.30308475 0.28496525 0.29907035</span></span>
<span><span class="co">## 3 cg26955540 0.20043204 0.17149777 0.21078054</span></span>
<span><span class="co">## 4 cg25170017 0.21122187 0.22894394 0.23500608</span></span>
<span><span class="co">## 5 cg12827637 0.13248314 0.13797303 0.14133260</span></span>
<span><span class="co">## 6 cg19442545 0.06102038 0.05626473 0.05382755</span></span></code></pre>
<p><code>sampleMeta</code> should include all sample names in
<code>samples</code>, and specify the origins they should be mapped to
when added to <code>atlas</code>.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">sampleMeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span><span class="st">"Experiment_accession"</span> <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">samples</span><span class="op">)</span><span class="op">[</span><span class="op">-</span><span class="fl">1</span><span class="op">]</span>,</span>
<span>                         <span class="st">"Biosample_term_name"</span> <span class="op">=</span> <span class="st">"new cell type"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">sampleMeta</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Experiment_accession Biosample_term_name</span></span>
<span><span class="co">## 1:             Sample 1       new cell type</span></span>
<span><span class="co">## 2:             Sample 2       new cell type</span></span>
<span><span class="co">## 3:             Sample 3       new cell type</span></span></code></pre>
<p>Use <code>findSignatures</code> to extend the matrix.</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">extended_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findSignatures.html">findSignatures</a></span><span class="op">(</span>samples <span class="op">=</span> <span class="va">samples</span>, </span>
<span>                                 sampleMeta <span class="op">=</span> <span class="va">sampleMeta</span>, </span>
<span>                                 atlas <span class="op">=</span> <span class="va">HumanCellTypeMethAtlas</span>,</span>
<span>                                 IDs <span class="op">=</span> <span class="st">"IDs"</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CELL TYPES IN EXTENDED ATLAS: </span></span>
<span><span class="co">## new cell type </span></span>
<span><span class="co">## Monocytes_EPIC </span></span>
<span><span class="co">## B.cells_EPIC </span></span>
<span><span class="co">## CD4T.cells_EPIC </span></span>
<span><span class="co">## NK.cells_EPIC </span></span>
<span><span class="co">## CD8T.cells_EPIC </span></span>
<span><span class="co">## Neutrophils_EPIC </span></span>
<span><span class="co">## Erythrocyte_progenitors </span></span>
<span><span class="co">## Adipocytes </span></span>
<span><span class="co">## Cortical_neurons </span></span>
<span><span class="co">## Hepatocytes </span></span>
<span><span class="co">## Lung_cells </span></span>
<span><span class="co">## Pancreatic_beta_cells </span></span>
<span><span class="co">## Pancreatic_acinar_cells </span></span>
<span><span class="co">## Pancreatic_duct_cells </span></span>
<span><span class="co">## Vascular_endothelial_cells </span></span>
<span><span class="co">## Colon_epithelial_cells </span></span>
<span><span class="co">## Left_atrium </span></span>
<span><span class="co">## Bladder </span></span>
<span><span class="co">## Breast </span></span>
<span><span class="co">## Head_and_neck_larynx </span></span>
<span><span class="co">## Kidney </span></span>
<span><span class="co">## Prostate </span></span>
<span><span class="co">## Thyroid </span></span>
<span><span class="co">## Upper_GI </span></span>
<span><span class="co">## Uterus_cervix</span></span></code></pre>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">extended_matrix</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          IDs new_cell_type Monocytes_EPIC B.cells_EPIC CD4T.cells_EPIC</span></span>
<span><span class="co">## 1 cg08169020    0.17443099         0.8866       0.2615          0.0149</span></span>
<span><span class="co">## 2 cg25913761    0.29570679         0.8363       0.2210          0.2816</span></span>
<span><span class="co">## 3 cg26955540    0.19423678         0.7658       0.0222          0.1492</span></span>
<span><span class="co">## 4 cg25170017    0.22505730         0.8861       0.5116          0.1021</span></span>
<span><span class="co">## 5 cg12827637    0.13726292         0.5212       0.3614          0.0227</span></span>
<span><span class="co">## 6 cg19442545    0.05703755         0.2013       0.1137          0.0608</span></span>
<span><span class="co">##   NK.cells_EPIC CD8T.cells_EPIC Neutrophils_EPIC Erythrocyte_progenitors</span></span>
<span><span class="co">## 1        0.0777          0.0164           0.8680                  0.9509</span></span>
<span><span class="co">## 2        0.4705          0.3961           0.8293                  0.2385</span></span>
<span><span class="co">## 3        0.4005          0.3474           0.7915                  0.1374</span></span>
<span><span class="co">## 4        0.4363          0.0875           0.7042                  0.9447</span></span>
<span><span class="co">## 5        0.2120          0.0225           0.5368                  0.4667</span></span>
<span><span class="co">## 6        0.0410          0.0668           0.1952                  0.1601</span></span>
<span><span class="co">##   Adipocytes Cortical_neurons Hepatocytes Lung_cells Pancreatic_beta_cells</span></span>
<span><span class="co">## 1     0.0336           0.0168      0.0340     0.0416              0.038875</span></span>
<span><span class="co">## 2     0.3578           0.3104      0.2389     0.2250              0.132000</span></span>
<span><span class="co">## 3     0.1965           0.0978      0.0338     0.0768              0.041725</span></span>
<span><span class="co">## 4     0.0842           0.2832      0.2259     0.0544              0.111750</span></span>
<span><span class="co">## 5     0.0287           0.1368      0.0307     0.1607              0.065975</span></span>
<span><span class="co">## 6     0.0364           0.0222      0.1574     0.0122              0.003825</span></span>
<span><span class="co">##   Pancreatic_acinar_cells Pancreatic_duct_cells Vascular_endothelial_cells</span></span>
<span><span class="co">## 1                  0.0209                0.0130                     0.0323</span></span>
<span><span class="co">## 2                  0.2249                0.1996                     0.3654</span></span>
<span><span class="co">## 3                  0.0314                0.0139                     0.2382</span></span>
<span><span class="co">## 4                  0.0309                0.0217                     0.0972</span></span>
<span><span class="co">## 5                  0.0370                0.0230                     0.0798</span></span>
<span><span class="co">## 6                  0.0378                0.0347                     0.0470</span></span>
<span><span class="co">##   Colon_epithelial_cells Left_atrium Bladder Breast Head_and_neck_larynx Kidney</span></span>
<span><span class="co">## 1                 0.0163      0.0386  0.0462 0.0264               0.0470 0.0269</span></span>
<span><span class="co">## 2                 0.2037      0.2446  0.2054 0.1922               0.2045 0.1596</span></span>
<span><span class="co">## 3                 0.0193      0.1134  0.1269 0.1651               0.1523 0.1034</span></span>
<span><span class="co">## 4                 0.0187      0.0674  0.0769 0.0691               0.0704 0.0604</span></span>
<span><span class="co">## 5                 0.0193      0.0432  0.0459 0.0228               0.0687 0.0234</span></span>
<span><span class="co">## 6                 0.0193      0.0287  0.0246 0.0081               0.0098 0.0309</span></span>
<span><span class="co">##   Prostate Thyroid Upper_GI Uterus_cervix</span></span>
<span><span class="co">## 1   0.0353  0.0553   0.0701        0.0344</span></span>
<span><span class="co">## 2   0.1557  0.1848   0.1680        0.2026</span></span>
<span><span class="co">## 3   0.0686  0.0943   0.1298        0.1075</span></span>
<span><span class="co">## 4   0.0369  0.0412   0.0924        0.0697</span></span>
<span><span class="co">## 5   0.0508  0.0726   0.0759        0.0196</span></span>
<span><span class="co">## 6   0.0055  0.0188   0.0090        0.0166</span></span></code></pre>
<p>WGBS methylation data first needs to be mapped to probes using
<code>BSmeth2Probe</code> before being deconvoluted. The methylation
data <code>WGBS_data</code> in <code>BSmeth2Probe</code> may be either a
<strong>GRanges</strong> object or a <strong>methylKit</strong>
object.</p>
<p>Format of <code>WGBS_data</code> as <strong>GRanges</strong>
object:</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/load.html" class="external-link">load</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"WGBS_GRanges.rda"</span>,</span>
<span>                                     package <span class="op">=</span> <span class="st">"deconvR"</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">WGBS_GRanges</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 6 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames    ranges strand |   sample1</span></span>
<span><span class="co">##          &lt;Rle&gt; &lt;IRanges&gt;  &lt;Rle&gt; | &lt;numeric&gt;</span></span>
<span><span class="co">##   [1]     chr1     47176      + |  0.833333</span></span>
<span><span class="co">##   [2]     chr1     47177      - |  0.818182</span></span>
<span><span class="co">##   [3]     chr1     47205      + |  0.681818</span></span>
<span><span class="co">##   [4]     chr1     47206      - |  0.583333</span></span>
<span><span class="co">##   [5]     chr1     47362      + |  0.416667</span></span>
<span><span class="co">##   [6]     chr1     49271      + |  0.733333</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 1 sequence from an unspecified genome; no seqlengths</span></span></code></pre>
<p>or as <strong>methylKit</strong> object:</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fu">methylKit</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/methylKit/man/methRead-methods.html" class="external-link">methRead</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"test1.myCpG.txt"</span>, </span>
<span>                                     package <span class="op">=</span> <span class="st">"methylKit"</span><span class="op">)</span>, </span>
<span>                         sample.id<span class="op">=</span><span class="st">"test"</span>, assembly<span class="op">=</span><span class="st">"hg18"</span>, </span>
<span>                         treatment<span class="op">=</span><span class="fl">1</span>, context<span class="op">=</span><span class="st">"CpG"</span>, mincov <span class="op">=</span> <span class="fl">0</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##     chr   start     end strand coverage numCs numTs</span></span>
<span><span class="co">## 1 chr21 9764513 9764513      -       12     0    12</span></span>
<span><span class="co">## 2 chr21 9764539 9764539      -       12     3     9</span></span>
<span><span class="co">## 3 chr21 9820622 9820622      +       13     0    13</span></span>
<span><span class="co">## 4 chr21 9837545 9837545      +       11     0    11</span></span>
<span><span class="co">## 5 chr21 9849022 9849022      +      124    90    34</span></span>
<span><span class="co">## 6 chr21 9853296 9853296      +       17    10     7</span></span></code></pre>
<p><code>probe_id_locations</code> contains information needed to map
cellular loci to probe IDs</p>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"IlluminaMethEpicB5ProbeIDs"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">IlluminaMethEpicB5ProbeIDs</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## GRanges object with 6 ranges and 1 metadata column:</span></span>
<span><span class="co">##       seqnames              ranges strand |          ID</span></span>
<span><span class="co">##          &lt;Rle&gt;           &lt;IRanges&gt;  &lt;Rle&gt; | &lt;character&gt;</span></span>
<span><span class="co">##   [1]    chr19     5236005-5236006      + |  cg07881041</span></span>
<span><span class="co">##   [2]    chr20   63216298-63216299      + |  cg18478105</span></span>
<span><span class="co">##   [3]     chr1     6781065-6781066      + |  cg23229610</span></span>
<span><span class="co">##   [4]     chr2 197438742-197438743      - |  cg03513874</span></span>
<span><span class="co">##   [5]     chrX   24054523-24054524      + |  cg09835024</span></span>
<span><span class="co">##   [6]    chr14   93114794-93114795      - |  cg05451842</span></span>
<span><span class="co">##   -------</span></span>
<span><span class="co">##   seqinfo: 25 sequences from an unspecified genome; no seqlengths</span></span></code></pre>
<p>Use <code>BSmeth2Probe</code> to map WGBS data to probe IDs.</p>
<div class="sourceCode" id="cb20"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mapped_WGBS_data</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/BSmeth2Probe.html">BSmeth2Probe</a></span><span class="op">(</span>probe_id_locations <span class="op">=</span> <span class="va">IlluminaMethEpicB5ProbeIDs</span>, </span>
<span>                                 WGBS_data <span class="op">=</span> <span class="va">WGBS_GRanges</span>,</span>
<span>                                 multipleMapping <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                                 cutoff <span class="op">=</span> <span class="fl">10</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">mapped_WGBS_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##          IDs   sample1</span></span>
<span><span class="co">## 1 cg00305774 1.0000000</span></span>
<span><span class="co">## 2 cg00546078 0.8181818</span></span>
<span><span class="co">## 3 cg00546307 0.5454545</span></span>
<span><span class="co">## 4 cg00546971 0.5000000</span></span>
<span><span class="co">## 5 cg00774867 0.8461538</span></span>
<span><span class="co">## 6 cg00830435 0.9166667</span></span></code></pre>
<p>This mapped data can now be used in <code>deconvolute</code>. Here we
will deconvolute it using the previously extended signature matrix as
the reference atlas.</p>
<div class="sourceCode" id="cb22"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deconvolution</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span>reference <span class="op">=</span> <span class="va">HumanCellTypeMethAtlas</span>, </span>
<span>                             bulk <span class="op">=</span> <span class="va">mapped_WGBS_data</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##    Min. 1st Qu.  Median    Mean 3rd Qu.    Max. </span></span>
<span><span class="co">##  0.9963  0.9963  0.9963  0.9963  0.9963  0.9963</span></span></code></pre>
<div class="sourceCode" id="cb24"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">deconvolution</span><span class="op">$</span><span class="va">proportions</span></span></code></pre></div>
<pre><code><span><span class="co">##         Monocytes_EPIC B.cells_EPIC CD4T.cells_EPIC NK.cells_EPIC</span></span>
<span><span class="co">## sample1              0            0               0             0</span></span>
<span><span class="co">##         CD8T.cells_EPIC Neutrophils_EPIC Erythrocyte_progenitors Adipocytes</span></span>
<span><span class="co">## sample1               0                0                       0  0.5011789</span></span>
<span><span class="co">##         Cortical_neurons Hepatocytes Lung_cells Pancreatic_beta_cells</span></span>
<span><span class="co">## sample1        0.2917357           0          0                     0</span></span>
<span><span class="co">##         Pancreatic_acinar_cells Pancreatic_duct_cells</span></span>
<span><span class="co">## sample1                       0                     0</span></span>
<span><span class="co">##         Vascular_endothelial_cells Colon_epithelial_cells Left_atrium Bladder</span></span>
<span><span class="co">## sample1                          0            0.001012524           0       0</span></span>
<span><span class="co">##         Breast Head_and_neck_larynx Kidney Prostate Thyroid Upper_GI</span></span>
<span><span class="co">## sample1      0            0.2060729      0        0       0        0</span></span>
<span><span class="co">##         Uterus_cervix</span></span>
<span><span class="co">## sample1             0</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="constructing-tissue-specific-cpg-signature-matrix">Constructing tissue specific CpG signature matrix<a class="anchor" aria-label="anchor" href="#constructing-tissue-specific-cpg-signature-matrix"></a>
</h3>
<p>Alternatively, users can set <em>tissueSpecCpGs</em> as
<strong>TRUE</strong> to construct tissue based methylation signature
matrix by using the reference atlas. Here, we used simulated samples to
construct tissue specific signature matrix since we don’t have tissue
specific samples.</p>
<div class="sourceCode" id="cb26"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HumanCellTypeMethAtlas"</span><span class="op">)</span></span>
<span><span class="va">exampleSamples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulateCellMix.html">simulateCellMix</a></span><span class="op">(</span><span class="fl">1</span>,reference <span class="op">=</span> <span class="va">HumanCellTypeMethAtlas</span><span class="op">)</span><span class="op">$</span><span class="va">simulated</span></span>
<span><span class="va">exampleMeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span><span class="st">"Experiment_accession"</span> <span class="op">=</span> <span class="st">"example_sample"</span>,</span>
<span>                          <span class="st">"Biosample_term_name"</span> <span class="op">=</span> <span class="st">"example_cell_type"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exampleSamples</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CpGs"</span>, <span class="st">"example_sample"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">HumanCellTypeMethAtlas</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CpGs"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">signatures</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findSignatures.html">findSignatures</a></span><span class="op">(</span></span>
<span>  samples <span class="op">=</span> <span class="va">exampleSamples</span>,</span>
<span>  sampleMeta <span class="op">=</span> <span class="va">exampleMeta</span>,</span>
<span>  atlas <span class="op">=</span> <span class="va">HumanCellTypeMethAtlas</span>,</span>
<span>  IDs <span class="op">=</span> <span class="st">"CpGs"</span>, K <span class="op">=</span> <span class="fl">100</span>, tissueSpecCpGs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CELL TYPES IN EXTENDED ATLAS: </span></span>
<span><span class="co">## example_cell_type </span></span>
<span><span class="co">## Monocytes_EPIC </span></span>
<span><span class="co">## B.cells_EPIC </span></span>
<span><span class="co">## CD4T.cells_EPIC </span></span>
<span><span class="co">## NK.cells_EPIC </span></span>
<span><span class="co">## CD8T.cells_EPIC </span></span>
<span><span class="co">## Neutrophils_EPIC </span></span>
<span><span class="co">## Erythrocyte_progenitors </span></span>
<span><span class="co">## Adipocytes </span></span>
<span><span class="co">## Cortical_neurons </span></span>
<span><span class="co">## Hepatocytes </span></span>
<span><span class="co">## Lung_cells </span></span>
<span><span class="co">## Pancreatic_beta_cells </span></span>
<span><span class="co">## Pancreatic_acinar_cells </span></span>
<span><span class="co">## Pancreatic_duct_cells </span></span>
<span><span class="co">## Vascular_endothelial_cells </span></span>
<span><span class="co">## Colon_epithelial_cells </span></span>
<span><span class="co">## Left_atrium </span></span>
<span><span class="co">## Bladder </span></span>
<span><span class="co">## Breast </span></span>
<span><span class="co">## Head_and_neck_larynx </span></span>
<span><span class="co">## Kidney </span></span>
<span><span class="co">## Prostate </span></span>
<span><span class="co">## Thyroid </span></span>
<span><span class="co">## Upper_GI </span></span>
<span><span class="co">## Uterus_cervix</span></span></code></pre>
<div class="sourceCode" id="cb28"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">signatures</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## $hyper</span></span>
<span><span class="co">## cg08169020 cg25170017 cg26955540 cg12827637 cg15838173 cg04858631 cg19442545 </span></span>
<span><span class="co">## 0.23711510 0.16804743 0.16706298 0.16341740 0.14756377 0.14636983 0.14431474 </span></span>
<span><span class="co">## cg10560079 cg00982136 cg26460530 cg22017733 cg13677741 cg24082121 cg09642825 </span></span>
<span><span class="co">## 0.14131002 0.13602910 0.12922689 0.12882796 0.12651448 0.12433428 0.12393559 </span></span>
<span><span class="co">## cg01424562 cg15376996 cg14789659 cg11895835 cg06340704 cg12474798 cg19300307 </span></span>
<span><span class="co">## 0.12348807 0.11995284 0.11677111 0.11557836 0.11538565 0.11514650 0.11512351 </span></span>
<span><span class="co">## cg05507654 cg14785464 cg27189395 cg18856478 cg26033520 cg05056497 cg24275356 </span></span>
<span><span class="co">## 0.11509691 0.11470086 0.11453378 0.11412062 0.11360042 0.11245422 0.11226979 </span></span>
<span><span class="co">## cg23220823 cg00936790 cg25913761 cg20684110 cg08474651 cg08857351 cg06167719 </span></span>
<span><span class="co">## 0.11222956 0.11212217 0.11185164 0.11171525 0.11150135 0.11138356 0.11132746 </span></span>
<span><span class="co">## cg27395200 cg05258935 cg10509607 cg22897141 cg23247274 cg22879098 cg14855367 </span></span>
<span><span class="co">## 0.11125977 0.11114588 0.11070342 0.10968420 0.10965287 0.10959565 0.10919888 </span></span>
<span><span class="co">## cg03254916 cg02962602 cg08063160 cg08832851 cg26953232 cg04913246 cg16416715 </span></span>
<span><span class="co">## 0.10902188 0.10886200 0.10885235 0.10854991 0.10849166 0.10826308 0.10807710 </span></span>
<span><span class="co">## cg09267773 cg10361922 cg08257293 cg27388962 cg08425810 cg03711944 cg26757820 </span></span>
<span><span class="co">## 0.10797153 0.10788173 0.10748562 0.10747892 0.10744440 0.10727008 0.10710562 </span></span>
<span><span class="co">## cg26047334 cg25158622 cg14057303 cg11975790 cg08367326 cg16278496 cg14845962 </span></span>
<span><span class="co">## 0.10696897 0.10695904 0.10694399 0.10655879 0.10570196 0.10501968 0.10485859 </span></span>
<span><span class="co">## cg05043794 cg16127573 cg09577804 cg15014975 cg27312961 cg04055490 cg18675610 </span></span>
<span><span class="co">## 0.10433473 0.10420635 0.10399857 0.10397506 0.10384322 0.10370734 0.10330938 </span></span>
<span><span class="co">## cg22138735 cg08846870 cg19331221 cg17847861 cg07865091 cg03063309 cg24591770 </span></span>
<span><span class="co">## 0.10296365 0.10293987 0.10270677 0.10256418 0.10214005 0.10213549 0.10187307 </span></span>
<span><span class="co">## cg16402452 cg06398251 cg13638257 cg07218880 cg11532431 cg01377358 cg07438103 </span></span>
<span><span class="co">## 0.10187169 0.10185237 0.10176875 0.10163880 0.10161528 0.10125056 0.10116898 </span></span>
<span><span class="co">## cg02297709 cg01726273 cg04462378 cg10661769 cg00769843 cg16829306 cg11025609 </span></span>
<span><span class="co">## 0.10110097 0.10098302 0.10093560 0.10079763 0.10075990 0.10070226 0.10044612 </span></span>
<span><span class="co">## cg21181453 cg16988611 cg06889535 cg15185986 cg20074395 cg16306706 cg15575375 </span></span>
<span><span class="co">## 0.10042446 0.10026995 0.10026243 0.10021530 0.10009661 0.09967501 0.09956515 </span></span>
<span><span class="co">## cg08659179 cg19692929 </span></span>
<span><span class="co">## 0.09956422 0.09937775 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## $hypo</span></span>
<span><span class="co">## cg03663120 cg20942286 cg03963853 cg03126713 cg00828556 cg24788483 cg13931640 </span></span>
<span><span class="co">##  0.3750316  0.2984432  0.2728312  0.2722571  0.2699192  0.2695167  0.2658525 </span></span>
<span><span class="co">## cg11186858 cg22528270 cg15871206 cg05612654 cg13500029 cg15310871 cg10480329 </span></span>
<span><span class="co">##  0.2649158  0.2631947  0.2629806  0.2625828  0.2582106  0.2578245  0.2494196 </span></span>
<span><span class="co">## cg11231069 cg03549146 cg12655112 cg05923857 cg12458039 cg07737292 cg20610950 </span></span>
<span><span class="co">##  0.2468815  0.2403339  0.2391589  0.2373173  0.2360880  0.2342849  0.2330547 </span></span>
<span><span class="co">## cg10456459 cg03313271 cg06988336 cg23952578 cg06517984 cg16636767 cg06297318 </span></span>
<span><span class="co">##  0.2292928  0.2239580  0.2209195  0.2149117  0.2138972  0.2134217  0.2120877 </span></span>
<span><span class="co">## cg27334271 cg14976569 cg04851465 cg11327657 cg11597902 cg15633603 cg22259797 </span></span>
<span><span class="co">##  0.2119242  0.2105749  0.2097883  0.2096965  0.2090253  0.2078216  0.2075520 </span></span>
<span><span class="co">## cg13403369 cg08425796 cg06373940 cg04354689 cg09322573 cg18990407 cg20107506 </span></span>
<span><span class="co">##  0.2059912  0.2056366  0.2037298  0.2033962  0.2026265  0.2022337  0.2021727 </span></span>
<span><span class="co">## cg22185977 cg06125903 cg07033722 cg10718056 cg26538782 cg12866960 cg01879591 </span></span>
<span><span class="co">##  0.2018702  0.2008651  0.2006998  0.2006361  0.1991279  0.1990950  0.1985201 </span></span>
<span><span class="co">## cg25517015 cg27366072 cg03310874 cg13093111 cg02796279 cg19502671 cg08538581 </span></span>
<span><span class="co">##  0.1979015  0.1978145  0.1963424  0.1954368  0.1952972  0.1945160  0.1940933 </span></span>
<span><span class="co">## cg06978145 cg20429104 cg26923863 cg10967114 cg23403750 cg26783127 cg06721411 </span></span>
<span><span class="co">##  0.1939280  0.1936473  0.1928298  0.1926123  0.1925248  0.1922389  0.1921110 </span></span>
<span><span class="co">## cg11153071 cg19380303 cg26305504 cg23755933 cg01024962 cg20966357 cg06585734 </span></span>
<span><span class="co">##  0.1920780  0.1917895  0.1917448  0.1906979  0.1906406  0.1902112  0.1901046 </span></span>
<span><span class="co">## cg11802666 cg05445326 cg07918933 cg12555086 cg04664897 cg25596405 cg04217515 </span></span>
<span><span class="co">##  0.1899845  0.1896992  0.1896795  0.1896435  0.1895192  0.1892883  0.1890853 </span></span>
<span><span class="co">## cg04836151 cg27340480 cg24250902 cg18835472 cg00009088 cg02244028 cg23334433 </span></span>
<span><span class="co">##  0.1886103  0.1885386  0.1883981  0.1880701  0.1875629  0.1875230  0.1873931 </span></span>
<span><span class="co">## cg26889953 cg26298914 cg14189391 cg22875823 cg17117981 cg13980609 cg08400494 </span></span>
<span><span class="co">##  0.1873594  0.1871591  0.1865980  0.1864033  0.1862525  0.1861065  0.1858963 </span></span>
<span><span class="co">## cg07417146 cg15288326 cg17086773 cg08428868 cg04586126 cg22662844 cg00235484 </span></span>
<span><span class="co">##  0.1856974  0.1856663  0.1854674  0.1854316  0.1849504  0.1842936  0.1834122 </span></span>
<span><span class="co">## cg22666015 cg01622606 </span></span>
<span><span class="co">##  0.1829954  0.1814323</span></span></code></pre>
</div>
<div class="section level3">
<h3 id="constructing-tissue-specific-dmps">Constructing tissue specific DMPs<a class="anchor" aria-label="anchor" href="#constructing-tissue-specific-dmps"></a>
</h3>
<p>Alternatively, users can set <em>tissueSpecDMPs</em> as
<strong>TRUE</strong> to obtain tissue based DMPs by using the reference
atlas. Here, we used simulated samples since we don’t have tissue
specific samples. Note that both <em>tissueSpecCpGs</em> and
<em>tissueSpecDMPs</em> can’t be <em>TRUE</em> at the same time.</p>
<div class="sourceCode" id="cb30"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/data.html" class="external-link">data</a></span><span class="op">(</span><span class="st">"HumanCellTypeMethAtlas"</span><span class="op">)</span></span>
<span><span class="va">exampleSamples</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/simulateCellMix.html">simulateCellMix</a></span><span class="op">(</span><span class="fl">1</span>,reference <span class="op">=</span> <span class="va">HumanCellTypeMethAtlas</span><span class="op">)</span><span class="op">$</span><span class="va">simulated</span></span>
<span><span class="va">exampleMeta</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://Rdatatable.gitlab.io/data.table/reference/data.table.html" class="external-link">data.table</a></span><span class="op">(</span><span class="st">"Experiment_accession"</span> <span class="op">=</span> <span class="st">"example_sample"</span>,</span>
<span>                          <span class="st">"Biosample_term_name"</span> <span class="op">=</span> <span class="st">"example_cell_type"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">exampleSamples</span><span class="op">)</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CpGs"</span>, <span class="st">"example_sample"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">colnames</a></span><span class="op">(</span><span class="va">HumanCellTypeMethAtlas</span><span class="op">)</span><span class="op">[</span><span class="fl">1</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"CpGs"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">signatures</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/findSignatures.html">findSignatures</a></span><span class="op">(</span></span>
<span>  samples <span class="op">=</span> <span class="va">exampleSamples</span>,</span>
<span>  sampleMeta <span class="op">=</span> <span class="va">exampleMeta</span>,</span>
<span>  atlas <span class="op">=</span> <span class="va">HumanCellTypeMethAtlas</span>,</span>
<span>  IDs <span class="op">=</span> <span class="st">"CpGs"</span>, tissueSpecDMPs <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## CELL TYPES IN EXTENDED ATLAS: </span></span>
<span><span class="co">## example_cell_type </span></span>
<span><span class="co">## Monocytes_EPIC </span></span>
<span><span class="co">## B.cells_EPIC </span></span>
<span><span class="co">## CD4T.cells_EPIC </span></span>
<span><span class="co">## NK.cells_EPIC </span></span>
<span><span class="co">## CD8T.cells_EPIC </span></span>
<span><span class="co">## Neutrophils_EPIC </span></span>
<span><span class="co">## Erythrocyte_progenitors </span></span>
<span><span class="co">## Adipocytes </span></span>
<span><span class="co">## Cortical_neurons </span></span>
<span><span class="co">## Hepatocytes </span></span>
<span><span class="co">## Lung_cells </span></span>
<span><span class="co">## Pancreatic_beta_cells </span></span>
<span><span class="co">## Pancreatic_acinar_cells </span></span>
<span><span class="co">## Pancreatic_duct_cells </span></span>
<span><span class="co">## Vascular_endothelial_cells </span></span>
<span><span class="co">## Colon_epithelial_cells </span></span>
<span><span class="co">## Left_atrium </span></span>
<span><span class="co">## Bladder </span></span>
<span><span class="co">## Breast </span></span>
<span><span class="co">## Head_and_neck_larynx </span></span>
<span><span class="co">## Kidney </span></span>
<span><span class="co">## Prostate </span></span>
<span><span class="co">## Thyroid </span></span>
<span><span class="co">## Upper_GI </span></span>
<span><span class="co">## Uterus_cervix</span></span></code></pre>
<div class="sourceCode" id="cb32"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">signatures</span><span class="op">[[</span><span class="fl">2</span><span class="op">]</span><span class="op">]</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">##            intercept        f         pval         qval</span></span>
<span><span class="co">## cg10480329 -3.287511 212.1121 2.059834e-13 1.241668e-09</span></span>
<span><span class="co">## cg06297318 -3.496336 175.2838 1.591426e-12 4.796558e-09</span></span>
<span><span class="co">## cg18835472 -3.350244 155.5288 5.608606e-12 1.126956e-08</span></span>
<span><span class="co">## cg05923857 -3.515739 149.1711 8.667363e-12 1.306172e-08</span></span>
<span><span class="co">## cg27366072 -2.858145 143.1169 1.332008e-11 1.380701e-08</span></span>
<span><span class="co">## cg15633603 -1.496293 142.6850 1.374287e-11 1.380701e-08</span></span></code></pre>
</div>
</div>
<div class="section level2">
<h2 id="example-workflow-for-rna-sequencing-data">Example Workflow For RNA Sequencing Data<a class="anchor" aria-label="anchor" href="#example-workflow-for-rna-sequencing-data"></a>
</h2>
<p>It is possible to use RNA-seq data for deconvolution via
<strong>deconvR</strong> package. Beware that you have to set
<code>IDs</code> column that contains <code>Gene names</code> to run
<strong>deconvR</strong> functions. Therefore you can simulate bulk
RNA-seq data via <code>simulateCellMix</code> and, extend RNA-seq
reference atlas via <code>findSignatures</code>.</p>
</div>
<div class="section level2">
<h2 id="sessioninfo">sessionInfo<a class="anchor" aria-label="anchor" href="#sessioninfo"></a>
</h2>
<div class="sourceCode" id="cb34"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/sessionInfo.html" class="external-link">sessionInfo</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<pre><code><span><span class="co">## R version 4.1.3 (2022-03-10)</span></span>
<span><span class="co">## Platform: x86_64-conda-linux-gnu (64-bit)</span></span>
<span><span class="co">## Running under: Ubuntu 20.04.3 LTS</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## Matrix products: default</span></span>
<span><span class="co">## BLAS/LAPACK: /home/igunduz/.conda/envs/rdev/lib/libopenblasp-r0.3.21.so</span></span>
<span><span class="co">## </span></span>
<span><span class="co">## locale:</span></span>
<span><span class="co">##  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              </span></span>
<span><span class="co">##  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    </span></span>
<span><span class="co">##  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   </span></span>
<span><span class="co">##  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 </span></span>
<span><span class="co">##  [9] LC_ADDRESS=C               LC_TELEPHONE=C            </span></span>
<span><span class="co">## [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## attached base packages:</span></span>
<span><span class="co">## [1] parallel  stats     graphics  grDevices utils     datasets  methods  </span></span>
<span><span class="co">## [8] base     </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## other attached packages:</span></span>
<span><span class="co">## [1] dplyr_1.0.10      doParallel_1.0.17 iterators_1.0.14  foreach_1.5.2    </span></span>
<span><span class="co">## [5] deconvR_1.4.4     data.table_1.14.2 knitr_1.40        BiocStyle_2.22.0 </span></span>
<span><span class="co">## </span></span>
<span><span class="co">## loaded via a namespace (and not attached):</span></span>
<span><span class="co">##   [1] rsq_2.5                     BiocFileCache_2.2.1        </span></span>
<span><span class="co">##   [3] systemfonts_1.0.4           plyr_1.8.7                 </span></span>
<span><span class="co">##   [5] methylKit_1.20.0            splines_4.1.3              </span></span>
<span><span class="co">##   [7] BiocParallel_1.28.3         GenomeInfoDb_1.30.1        </span></span>
<span><span class="co">##   [9] ggplot2_3.3.6               digest_0.6.29              </span></span>
<span><span class="co">##  [11] htmltools_0.5.3             fansi_1.0.3                </span></span>
<span><span class="co">##  [13] magrittr_2.0.3              memoise_2.0.1              </span></span>
<span><span class="co">##  [15] tzdb_0.3.0                  limma_3.50.3               </span></span>
<span><span class="co">##  [17] readr_2.1.2                 Biostrings_2.62.0          </span></span>
<span><span class="co">##  [19] annotate_1.72.0             matrixStats_0.62.0         </span></span>
<span><span class="co">##  [21] R.utils_2.12.0              askpass_1.1                </span></span>
<span><span class="co">##  [23] bdsmatrix_1.3-6             pkgdown_2.0.6              </span></span>
<span><span class="co">##  [25] siggenes_1.68.0             prettyunits_1.1.1          </span></span>
<span><span class="co">##  [27] colorspace_2.0-3            rappdirs_0.3.3             </span></span>
<span><span class="co">##  [29] blob_1.2.3                  textshaping_0.3.6          </span></span>
<span><span class="co">##  [31] xfun_0.32                   crayon_1.5.1               </span></span>
<span><span class="co">##  [33] RCurl_1.98-1.8              jsonlite_1.8.0             </span></span>
<span><span class="co">##  [35] lme4_1.1-31                 genefilter_1.76.0          </span></span>
<span><span class="co">##  [37] GEOquery_2.62.2             survival_3.4-0             </span></span>
<span><span class="co">##  [39] glue_1.6.2                  gtable_0.3.1               </span></span>
<span><span class="co">##  [41] nnls_1.4                    zlibbioc_1.40.0            </span></span>
<span><span class="co">##  [43] XVector_0.34.0              DelayedArray_0.20.0        </span></span>
<span><span class="co">##  [45] Rhdf5lib_1.16.0             BiocGenerics_0.40.0        </span></span>
<span><span class="co">##  [47] HDF5Array_1.22.1            scales_1.2.1               </span></span>
<span><span class="co">##  [49] mvtnorm_1.1-3               rngtools_1.5.2             </span></span>
<span><span class="co">##  [51] DBI_1.1.3                   Rcpp_1.0.9                 </span></span>
<span><span class="co">##  [53] progress_1.2.2              xtable_1.8-4               </span></span>
<span><span class="co">##  [55] emdbook_1.3.12              bumphunter_1.36.0          </span></span>
<span><span class="co">##  [57] bit_4.0.4                   proxy_0.4-27               </span></span>
<span><span class="co">##  [59] mclust_6.0.0                preprocessCore_1.56.0      </span></span>
<span><span class="co">##  [61] stats4_4.1.3                httr_1.4.4                 </span></span>
<span><span class="co">##  [63] RColorBrewer_1.1-3          ellipsis_0.3.2             </span></span>
<span><span class="co">##  [65] pkgconfig_2.0.3             reshape_0.8.9              </span></span>
<span><span class="co">##  [67] XML_3.99-0.10               R.methodsS3_1.8.2          </span></span>
<span><span class="co">##  [69] dbplyr_2.2.1                sass_0.4.2                 </span></span>
<span><span class="co">##  [71] locfit_1.5-9.6              utf8_1.2.2                 </span></span>
<span><span class="co">##  [73] tidyselect_1.1.2            rlang_1.0.5                </span></span>
<span><span class="co">##  [75] reshape2_1.4.4              AnnotationDbi_1.56.2       </span></span>
<span><span class="co">##  [77] munsell_0.5.0               tools_4.1.3                </span></span>
<span><span class="co">##  [79] cachem_1.0.6                cli_3.4.0                  </span></span>
<span><span class="co">##  [81] generics_0.1.3              RSQLite_2.2.16             </span></span>
<span><span class="co">##  [83] fastseg_1.40.0              evaluate_0.16              </span></span>
<span><span class="co">##  [85] stringr_1.4.1               fastmap_1.1.0              </span></span>
<span><span class="co">##  [87] yaml_2.3.5                  ragg_1.2.2                 </span></span>
<span><span class="co">##  [89] bit64_4.0.5                 fs_1.5.2                   </span></span>
<span><span class="co">##  [91] beanplot_1.3.1              scrime_1.3.5               </span></span>
<span><span class="co">##  [93] purrr_0.3.4                 KEGGREST_1.34.0            </span></span>
<span><span class="co">##  [95] nlme_3.1-159                doRNG_1.8.2                </span></span>
<span><span class="co">##  [97] sparseMatrixStats_1.6.0     nor1mix_1.3-0              </span></span>
<span><span class="co">##  [99] R.oo_1.25.0                 xml2_1.3.3                 </span></span>
<span><span class="co">## [101] biomaRt_2.50.3              compiler_4.1.3             </span></span>
<span><span class="co">## [103] filelock_1.0.2              curl_4.3.2                 </span></span>
<span><span class="co">## [105] png_0.1-7                   e1071_1.7-12               </span></span>
<span><span class="co">## [107] tibble_3.1.8                bslib_0.4.0                </span></span>
<span><span class="co">## [109] stringi_1.7.8               GenomicFeatures_1.46.5     </span></span>
<span><span class="co">## [111] desc_1.4.1                  minfi_1.40.0               </span></span>
<span><span class="co">## [113] lattice_0.20-45             Matrix_1.5-1               </span></span>
<span><span class="co">## [115] nloptr_2.0.3                multtest_2.50.0            </span></span>
<span><span class="co">## [117] vctrs_0.4.1                 pillar_1.8.1               </span></span>
<span><span class="co">## [119] lifecycle_1.0.2             rhdf5filters_1.6.0         </span></span>
<span><span class="co">## [121] BiocManager_1.30.18         jquerylib_0.1.4            </span></span>
<span><span class="co">## [123] bitops_1.0-7                rtracklayer_1.54.0         </span></span>
<span><span class="co">## [125] GenomicRanges_1.46.1        qvalue_2.26.0              </span></span>
<span><span class="co">## [127] R6_2.5.1                    BiocIO_1.4.0               </span></span>
<span><span class="co">## [129] bookdown_0.30               IRanges_2.28.0             </span></span>
<span><span class="co">## [131] codetools_0.2-18            boot_1.3-28                </span></span>
<span><span class="co">## [133] MASS_7.3-58.1               gtools_3.9.3               </span></span>
<span><span class="co">## [135] assertthat_0.2.1            rhdf5_2.38.1               </span></span>
<span><span class="co">## [137] SummarizedExperiment_1.24.0 openssl_2.0.2              </span></span>
<span><span class="co">## [139] rprojroot_2.0.3             rjson_0.2.21               </span></span>
<span><span class="co">## [141] Deriv_4.1.3                 GenomicAlignments_1.30.0   </span></span>
<span><span class="co">## [143] Rsamtools_2.10.0            S4Vectors_0.32.4           </span></span>
<span><span class="co">## [145] GenomeInfoDbData_1.2.7      hms_1.1.2                  </span></span>
<span><span class="co">## [147] quadprog_1.5-8              grid_4.1.3                 </span></span>
<span><span class="co">## [149] minqa_1.2.5                 tidyr_1.2.1                </span></span>
<span><span class="co">## [151] coda_0.19-4                 base64_2.0.1               </span></span>
<span><span class="co">## [153] class_7.3-20                rmarkdown_2.16             </span></span>
<span><span class="co">## [155] DelayedMatrixStats_1.16.0   illuminaio_0.36.0          </span></span>
<span><span class="co">## [157] MatrixGenerics_1.6.0        bbmle_1.0.25               </span></span>
<span><span class="co">## [159] numDeriv_2016.8-1.1         Biobase_2.54.0             </span></span>
<span><span class="co">## [161] restfulr_0.0.15</span></span></code></pre>
<div class="sourceCode" id="cb36"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">stopCluster</span><span class="op">(</span><span class="va">cl</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="footnotes footnotes-end-of-document">
<hr>
<ol>
<li id="fn1"><p>Moss, J. et al. (2018). Comprehensive human cell-type
methylation atlas reveals origins of circulating cell-free DNA in health
and disease. Nature communications, 9(1), 1-12. <a href="https://doi.org/10.1038/s41467-018-07466-6" class="external-link uri">https://doi.org/10.1038/s41467-018-07466-6</a><a href="#fnref1" class="footnote-back">↩︎</a></p></li>
</ol>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p></p>
<p>Developed by .</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
